lib/internal/modules/esm/translators.js

```diff
@@ -301,54 +307,60 @@ function createCJSNoSourceModuleWrap(url, isMain) {
   }, module);
 }
 
-translators.set('commonjs-sync', function requireCommonJS(url, source, isMain) {
+translators.set('commonjs-sync', function requireCommonJS(url, translateContext, parentURL) {
   initCJSParseSync();
 
-  return createCJSModuleWrap(url, source, isMain, 'commonjs', (module, source, url, filename, isMain) => {
-    assert(module === CJSModule._cache[filename]);
-    wrapModuleLoad(filename, null, isMain);
-  });
+  return createCJSModuleWrap(url, translateContext, parentURL, loadCJSModuleWithModuleLoad);
 });
 
 // Handle CommonJS modules referenced by `require` calls.
 // This translator function must be sync, as `require` is sync.
-translators.set('require-commonjs', (url, source, isMain) => {
+translators.set('require-commonjs', (url, translateContext, parentURL) => {
   initCJSParseSync();
   assert(cjsParse);
 
-  return createCJSModuleWrap(url, source, isMain, 'commonjs');
+  return createCJSModuleWrap(url, translateContext, parentURL);
 });
 
 // Handle CommonJS modules referenced by `require` calls.
 // This translator function must be sync, as `require` is sync.
-translators.set('require-commonjs-typescript', (url, source, isMain) => {
+translators.set('require-commonjs-typescript', (url, translateContext, parentURL) => {
   assert(cjsParse);
-  const code = stripTypeScriptModuleTypes(stringify(source), url);
-  return createCJSModuleWrap(url, code, isMain, 'commonjs-typescript');
+  translateContext.source = stripTypeScriptModuleTypes(stringify(translateContext.source), url);
+  return createCJSModuleWrap(url, translateContext, parentURL);
 });
 
+// This goes through Module._load to accommodate monkey-patchers.
+function loadCJSModuleWithModuleLoad(module, source, url, filename, isMain) {
+  assert(module === CJSModule._cache[filename]);
+  wrapModuleLoad(filename, undefined, isMain);
+}
+
 // Handle CommonJS modules referenced by `import` statements or expressions,
 // or as the initial entry point when the ESM loader handles a CommonJS entry.
-translators.set('commonjs', function commonjsStrategy(url, source, isMain) {
+translators.set('commonjs', function commonjsStrategy(url, translateContext, parentURL) {
   if (!cjsParse) {
     initCJSParseSync();
   }
 
   // For backward-compatibility, it's possible to return a nullish value for
-  // CJS source associated with a file: URL. In this case, the source is
-  // obtained by calling the monkey-patchable CJS loader.
-  const cjsLoader = source == null ? (module, source, url, filename, isMain) => {
-    assert(module === CJSModule._cache[filename]);
-    wrapModuleLoad(filename, undefined, isMain);
-  } : loadCJSModule;
+  // CJS source associated with a file: URL - that usually means the source is not
```

Comment:
nit (I see it was like that before): the colon makes it read like an explanation is following. I had to re-read the sentence a couple times to realise it was talking about a `file:` protocoled URL.

```suggestion
  // CJS source associated with a `file:` URL - that usually means the source is not
```